src/Storages/IStorage.h

```diff
@@ -221,6 +221,11 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
         /// proton: FIXME, introduce versioning schema physically in metastore
         return metadata.get();
     }
+
+    /// Insert queries squash blocks for buffering (see InterpreterInsertQuery.cpp).
+    /// However, not all storages need this feature, for example, external streams likely have
+    /// their own ways to do buffering.
+    virtual bool insertWithoutSquash() const { return false; }
```

Comment:
it seems better flip the logic here since by default we will need squash buffer and extern stream Kafka overrides it

```
virtual bool squashInsert() const noexcept { return true; }
```